code/physioconv/write_event_file.py

```diff
@@ -0,0 +1,247 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+import argparse
+import pandas as pd
+import os
+import gzip
+import json
+import matplotlib.pyplot as plt
+import matplotlib.patches as mpatches
+
+EVENTS_JSON_BOILERPLATE = {
+    "StimulusPresentation": {
+        "OperatingSystem": "Linux Ubuntu 20.04.5",
+        "SoftwareName": "PsychoPy",
+        "SoftwareRRID": "SCR_006571",
+        "SoftwareVersion": "2022.3.0.dev6"
+    }
+}
+
+def plot_physio_data_with_events(
+    time_series_df: pd.DataFrame,
+    events_df: pd.DataFrame,
+    tsv_file: str,
+    output_folder: str = ".",
+) -> None:
+    """
+    Plot physiological data along with events.
+
+    Args:
+        time_series_df (pd.DataFrame): DataFrame containing the physiological data.
+        events_df (pd.DataFrame): DataFrame containing the event data.
+        tsv_file (str): Path to the TSV file.
+        output_folder (str, optional): Output folder for saving the plot image. Default is current directory.
+
+    Returns:
+        None: The plot is saved as an PNG file.
+
+    """
+    fig, (ax1, ax2, ax3) = plt.subplots(3, 1, figsize=(30, 10), sharex=True)
+    ax1.plot(time_series_df[0], time_series_df[1], label="RB")
+    ax2.plot(time_series_df[0], time_series_df[2], label="ECG")
+    ax3.plot(time_series_df[0], time_series_df[3], label="GA")
+    trial_types = events_df["trial-type"].unique()
+    legend_colors = plt.cm.get_cmap("Set1", len(trial_types))
+    legend_color_dict = {
+        trial_type: legend_colors(i) for i, trial_type in enumerate(trial_types)
+    }
+    color_patches = []
+
+    for trial_type in trial_types:
+        legend_color = legend_color_dict[trial_type]
+        color_patch = mpatches.Patch(color=legend_color, label=trial_type)
+        color_patches.append(color_patch)
+        events_of_type = events_df[events_df["trial-type"] == trial_type]
+
+        for index, row in events_of_type.iterrows():
+            ax1.axvline(row["onset"], color=legend_color, linestyle="--")
+            ax2.axvline(row["onset"], color=legend_color, linestyle="--")
+            ax3.axvline(row["onset"], color=legend_color, linestyle="--")
+    ax1.set_ylabel("V")
+    ax2.set_ylabel("V")
+    ax3.set_ylabel("V")
+    ax3.legend(handles=color_patches, loc="lower right")
+    ax1.set_title("RB")
+    ax2.set_title("ECG")
+    ax3.set_title("GA")
+    base_name = os.path.basename(tsv_file)
+    output_file = os.path.join(
+        output_folder, base_name.replace("_physio.tsv.gz", "_plot.png")
+    )
+    plt.tight_layout()
+    plt.savefig(output_file)
+
```

Comment:
```suggestion
    return output_file

```

Reply:
I accepted the suggestion, but what is the idea behind having only this function return a value?

Reply:
What's the point of returning `None`?

I would honestly return the matplotlib figure object and save the figure outside the function, but returning the filename seemed to me like the least impactful change giving some extra value to the function.